test/gtest/ucp/test_ucp_worker.cc

```diff
@@ -264,7 +275,39 @@ class test_ucp_worker_discard : public ucp_test {
         disconnect(sender());
     }
 
-    static void ep_destroy_func(uct_ep_h ep) {
+    class flush_uct_ep_req_search {
+    public:
+        flush_uct_ep_req_search(uct_ep_h uct_ep) : m_uct_ep(uct_ep)
+        {
+        }
+
+        bool operator()(uct_completion_t *comp)
+        {
+            ucp_request_t *req = ucs_container_of(comp, ucp_request_t,
+                                                  send.state.uct_comp);
+            return req->send.discard_uct_ep.uct_ep == m_uct_ep;
+        }
+
+    private:
+        uct_ep_h m_uct_ep;
+    };
+
+    static void ep_destroy_func(uct_ep_h ep)
+    {
+        std::vector<uct_completion_t*>::iterator iter =
+                std::find_if(m_flush_comps.begin(), m_flush_comps.end(),
+                             flush_uct_ep_req_search(ep));
+
+        if (iter != m_flush_comps.end()) {
+            ucp_request_t *req = ucs_container_of(*iter, ucp_request_t,
+                                                  send.state.uct_comp);
+
+            req->send.state.uct_comp.count = 0;
+            req->send.state.uct_comp.func(*iter);
+
+            m_flush_comps.erase(iter);
+        }
```

Comment:
1. why ep_destroy calls comp function? do we have such behavior in UCT?
2. pls simply search&destroy. no need for a functor.

Reply:
> * why ep_destroy calls comp function? do we have such behavior in UCT?

yes, UCT transports (e.g. DC/RC/UD) alls completion functions in UCT EP destroy calls

> pls simply search&destroy. no need for a functor.

done